src/test/java/com/znsio/sample/e2e/screen/amazonsearch/AmazonProductScreen.java

```diff
@@ -0,0 +1,42 @@
+package com.znsio.sample.e2e.screen.amazonsearch;
+
+import com.znsio.e2e.entities.Platform;
+import com.znsio.e2e.runner.Runner;
+import com.znsio.e2e.tools.Driver;
+import com.znsio.e2e.tools.Visual;
+import com.znsio.sample.e2e.screen.android.amazonsearch.AmazonProductScreenAndroid;
+import com.znsio.sample.e2e.screen.web.amazonsearch.AmazonProductScreenWeb;
+import org.apache.commons.lang3.NotImplementedException;
+import org.apache.log4j.Logger;
+
+import static com.znsio.e2e.runner.Runner.fetchDriver;
+import static com.znsio.e2e.runner.Runner.fetchEyes;
+
+public abstract class AmazonProductScreen {
+    private static final String SCREEN_NAME = AmazonProductScreen.class.getSimpleName();
+    private static final Logger LOGGER = Logger.getLogger(SCREEN_NAME);
+
+    public static AmazonProductScreen get() {
+        Driver driver = fetchDriver(Thread.currentThread()
+                .getId());
+        Platform platform = Runner.fetchPlatform(Thread.currentThread()
+                .getId());
+        LOGGER.info(SCREEN_NAME + ": Driver type: " + driver.getType() + ": Platform: " + platform);
+        Visual visually = fetchEyes(Thread.currentThread()
+                .getId());
+
+        switch(platform) {
+            case android:
+                return new AmazonProductScreenAndroid(driver, visually);
+            case web:
+                return new AmazonProductScreenWeb(driver, visually);
+        }
+        throw new NotImplementedException(SCREEN_NAME + " is not implemented in " + Runner.platform);
+    }
+
+    public abstract boolean isProductDetailsDisplayed();
+
+    public abstract AmazonProductScreen changeToNewTab();
```

Comment:
do we really require a new method for switching the tab

Reply:
In web screen tab is changing after selecting the product. As view needs to be changed that is why a spearate fucntion is created. Otherwise it could be done in the same function also.